src/reconcile/rec_row.c

```diff
@@ -784,28 +784,43 @@ __wt_rec_row_leaf(
 
         /* Look for an update. */
         WT_ERR(__wt_rec_upd_select(session, r, NULL, rip, vpack, &upd_select));
-        if ((upd = upd_select.upd) != NULL) {
+        upd = upd_select.upd;
+
+        /*
+         * Figure out the timestamps. If there's no update and salvaging the file, clear the time
+         * pair information, else take the time pairs from the cell.
+         */
+        if (upd == NULL) {
+            if (!salvage) {
+                start_durable_ts = vpack->durable_start_ts;
+                start_ts = vpack->start_ts;
+                start_txn = vpack->start_txn;
+                stop_durable_ts = vpack->durable_stop_ts;
+                stop_ts = vpack->stop_ts;
+                stop_txn = vpack->stop_txn;
+            } else {
+                start_durable_ts = WT_TS_NONE;
```

Comment:
It feels like we should have a shorthand for this - though that might be hard since this is a set of local variables. Maybe that's weird in itself? Should there be a WT_TIME_WINDOW data structure that we use instead of declaring 6 different local variables?

That feels orthogonal to this change - so feel free to push ahead as-is

Reply:
Yeah, I've wondered the same thing (enough to futz with it once for a few minutes). I didn't come up with anything better than a macro that initialized its arguments, and that didn't feel much better. It would protect against fat-fingering the values, I guess.